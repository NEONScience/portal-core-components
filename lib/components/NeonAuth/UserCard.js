"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireDefault(require("react"));
var _Typography = _interopRequireDefault(require("@material-ui/core/Typography"));
var _Divider = _interopRequireDefault(require("@material-ui/core/Divider"));
var _Chip = _interopRequireDefault(require("@material-ui/core/Chip"));
var _Avatar = _interopRequireDefault(require("@material-ui/core/Avatar"));
var _Email = _interopRequireDefault(require("@material-ui/icons/Email"));
var _styles = require("@material-ui/core/styles");
var _moment = _interopRequireDefault(require("moment"));
var _Theme = _interopRequireDefault(require("../Theme/Theme"));
var _typeUtil = require("../../util/typeUtil");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
/* eslint-disable jsx-a11y/label-has-associated-control */

// eslint-disable-next-line no-unused-vars
const useStyles = (0, _styles.makeStyles)(theme =>
// eslint-disable-next-line implicit-arrow-linebreak
(0, _styles.createStyles)({
  cardHeaderContainer: {
    padding: '10px',
    '& label.account-title': {
      marginLeft: '20px',
      height: '68px',
      marginTop: '25px',
      fontWeight: 600,
      fontSize: '20px'
    },
    '& img.profile-image': {
      width: '120px'
    }
  },
  card: {
    margin: 'auto',
    textAlign: 'center'
  },
  profileArea: {
    '& img': {
      marginBottom: '15px',
      borderRadius: '50%',
      marginLeft: '10px',
      marginRight: '10px'
    }
  },
  profileInfoArea: {
    width: '100%',
    textAlign: 'left',
    display: 'inline-block',
    marginTop: '20px',
    '& i.fa': {
      marginRight: '10px'
    },
    '& i.fa-user': {
      verticalAlign: 'middle'
    },
    '& label': {
      fontSize: '16px',
      fontWeight: 600
    },
    '& label.label-block': {
      display: 'block',
      paddingBottom: '5px'
    },
    '& div.label-info-container': {
      padding: '10px',
      paddingTop: '20px',
      paddingLeft: '20px',
      paddingRight: '20px'
    },
    '& div.label-name-container': {
      padding: '10px'
    },
    '& label.username-label': {
      verticalAlign: 'middle'
    },
    '& label.label-normal': {
      fontSize: '14px',
      fontWeight: 'normal',
      wordBreak: 'break-all'
    }
  },
  emailChipContainer: {
    '& .MuiChip-root': {
      height: '100% !important'
    },
    '& .MuiChip-label': {
      width: '100% !important',
      wordBreak: 'break-all !important',
      whiteSpace: 'normal !important',
      padding: '6px'
    },
    '& .chip-email.MuiChip-root': {
      fontSize: '.90em !important'
    },
    '& .chip-email .MuiAvatar-root.MuiChip-avatar.MuiChip-avatarColorPrimary': {
      width: '32px !important',
      height: '32px !important'
    }
  }
}));
const UserCard = props => {
  const {
    pictureUrl,
    email,
    fullName,
    providers,
    lastLogin
  } = props;
  const classes = useStyles(_Theme.default);
  let lastLoginDisplay = null;
  if ((0, _typeUtil.exists)(lastLogin)) {
    const dateFmt = 'MMMM Do, YYYY h:mm:ss a';
    const lastLoginDate = (0, _moment.default)(new Date(lastLogin));
    lastLoginDisplay = lastLoginDate.format(dateFmt);
  }
  return /*#__PURE__*/_react.default.createElement("div", {
    className: classes.profileArea
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: classes.card
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: classes.cardHeaderContainer
  }, /*#__PURE__*/_react.default.createElement("img", {
    src: pictureUrl,
    className: "profile-image",
    alt: "Profile"
  })), /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement(_Typography.default, {
    variant: "h4",
    component: "h4",
    style: {
      margin: '10px'
    }
  }, fullName)), /*#__PURE__*/_react.default.createElement("div", {
    className: classes.emailChipContainer
  }, /*#__PURE__*/_react.default.createElement(_Chip.default, {
    avatar: /*#__PURE__*/_react.default.createElement(_Avatar.default, {
      color: "primary"
    }, /*#__PURE__*/_react.default.createElement(_Email.default, null)),
    key: email,
    label: email,
    className: "chip-email",
    color: "primary"
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: classes.profileInfoArea
  }, /*#__PURE__*/_react.default.createElement(_Divider.default, null), /*#__PURE__*/_react.default.createElement("div", {
    className: "label-info-container"
  }, /*#__PURE__*/_react.default.createElement("label", {
    className: "control-label label-block"
  }, "Identity Provider"), /*#__PURE__*/_react.default.createElement("label", {
    className: "label-normal"
  }, providers)), /*#__PURE__*/_react.default.createElement("div", {
    className: "label-info-container"
  }, /*#__PURE__*/_react.default.createElement("label", {
    className: "control-label label-block"
  }, "Last Login"), /*#__PURE__*/_react.default.createElement("label", {
    className: "label-normal"
  }, lastLoginDisplay)))));
};
var _default = exports.default = UserCard;