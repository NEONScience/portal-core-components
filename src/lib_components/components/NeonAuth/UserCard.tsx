/* eslint-disable jsx-a11y/label-has-associated-control */

import React from 'react';

import Typography from '@material-ui/core/Typography';
import Divider from '@material-ui/core/Divider';
import Chip from '@material-ui/core/Chip';
import Avatar from '@material-ui/core/Avatar';
import EmailIcon from '@material-ui/icons/Email';
import {
  makeStyles,
  createStyles,
  Theme as MuiThemeType,
} from '@material-ui/core/styles';

import moment, { Moment } from 'moment';

import Theme from '../Theme/Theme';
import { StringPropsObject, StylesHook } from '../../types/core';
import { exists } from '../../util/typeUtil';

// eslint-disable-next-line no-unused-vars
const useStyles: StylesHook = makeStyles((theme: MuiThemeType) =>
  // eslint-disable-next-line implicit-arrow-linebreak
  createStyles({
    cardHeaderContainer: {
      padding: '10px',
      '& label.account-title': {
        marginLeft: '20px',
        height: '68px',
        marginTop: '25px',
        fontWeight: 600,
        fontSize: '20px',
      },
      '& img.profile-image': {
        width: '120px',
      },
    },
    card: {
      margin: 'auto',
      textAlign: 'center',
    },
    profileArea: {
      '& img': {
        marginBottom: '15px',
        borderRadius: '50%',
        marginLeft: '10px',
        marginRight: '10px',
      },
    },
    profileInfoArea: {
      width: '100%',
      textAlign: 'left',
      display: 'inline-block',
      marginTop: '20px',
      '& i.fa': {
        marginRight: '10px',
      },
      '& i.fa-user': {
        verticalAlign: 'middle',
      },
      '& label': {
        fontSize: '16px',
        fontWeight: 600,
      },
      '& label.label-block': {
        display: 'block',
        paddingBottom: '5px',
      },
      '& div.label-info-container': {
        padding: '10px',
        paddingTop: '20px',
        paddingLeft: '20px',
        paddingRight: '20px',
      },
      '& div.label-name-container': {
        padding: '10px',
      },
      '& label.username-label': {
        verticalAlign: 'middle',
      },
      '& label.label-normal': {
        fontSize: '14px',
        fontWeight: 'normal',
        wordBreak: 'break-all',
      },
    },
    emailChipContainer: {
      '& .MuiChip-root': {
        height: '100% !important',
      },
      '& .MuiChip-label': {
        width: '100% !important',
        wordBreak: 'break-all !important',
        whiteSpace: 'normal !important',
        padding: '6px',
      },
      '& .chip-email.MuiChip-root': {
        fontSize: '.90em !important',
      },
      '& .chip-email .MuiAvatar-root.MuiChip-avatar.MuiChip-avatarColorPrimary': {
        width: '32px !important',
        height: '32px !important',
      },
    },
  })) as StylesHook;

export interface UserCardProps {
  pictureUrl: string;
  email: string;
  fullName: string;
  providers: string;
  // eslint-disable-next-line react/require-default-props
  lastLogin?: string;
}

const UserCard = (props: UserCardProps): JSX.Element => {
  const {
    pictureUrl,
    email,
    fullName,
    providers,
    lastLogin,
  }: UserCardProps = props;
  const classes: StringPropsObject = useStyles(Theme);

  let lastLoginDisplay: string | null = null;
  if (exists(lastLogin)) {
    const dateFmt: string = 'MMMM Do, YYYY h:mm:ss a';
    const lastLoginDate: Moment = moment(new Date(lastLogin as string));
    lastLoginDisplay = lastLoginDate.format(dateFmt);
  }

  return (
    <div className={classes.profileArea}>
      <div className={classes.card}>
        <div className={classes.cardHeaderContainer}>
          <img src={pictureUrl} className="profile-image" alt="Profile" />
        </div>
        <div>
          <Typography variant="h4" component="h4" style={{ margin: '10px' }}>
            {fullName}
          </Typography>
        </div>
        <div className={classes.emailChipContainer}>
          <Chip
            avatar={(
              <Avatar color="primary">
                <EmailIcon />
              </Avatar>
            )}
            key={email}
            label={email}
            className="chip-email"
            color="primary"
          />
        </div>
        <div className={classes.profileInfoArea}>
          <Divider />
          <div className="label-info-container">
            <label className="control-label label-block">
              Identity Provider
            </label>
            <label className="label-normal">{providers}</label>
          </div>
          <div className="label-info-container">
            <label className="control-label label-block">
              Last Login
            </label>
            <label className="label-normal">{lastLoginDisplay}</label>
          </div>
        </div>
      </div>
    </div>
  );
};

export default UserCard;
