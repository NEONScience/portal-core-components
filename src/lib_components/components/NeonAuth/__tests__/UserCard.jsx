import React from 'react';
import renderer from 'react-test-renderer';

import UserCard from '../UserCard';

describe('NeonAuth - UserCard', () => {
  test('Renders without a lastLogin prop', () => {
    const tree = renderer.create(
      <UserCard
        fullName="Guy Incognito"
        email="joeyjojojr@shabadu"
        providers="momar"
        pictureUrl="carhole"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Renders with a lastLogin prop', () => {
    const tree = renderer.create(
      <UserCard
        fullName="Slappy White"
        email="separate@knob"
        providers="monks"
        pictureUrl="label/baby/jr"
        lastLogin="2018-05-05T00:00:00Z"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
