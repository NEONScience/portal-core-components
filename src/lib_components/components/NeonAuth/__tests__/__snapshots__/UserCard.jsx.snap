// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NeonAuth - UserCard Renders with a lastLogin prop 1`] = `
<div
  className="makeStyles-profileArea-3"
>
  <div
    className="makeStyles-card-2"
  >
    <div
      className="makeStyles-cardHeaderContainer-1"
    >
      <img
        alt="Profile"
        className="profile-image"
        src="label/baby/jr"
      />
    </div>
    <div>
      <h4
        className="MuiTypography-root MuiTypography-h4"
        style={
          Object {
            "margin": "10px",
          }
        }
      >
        Slappy White
      </h4>
    </div>
    <div
      className="makeStyles-emailChipContainer-5"
    >
      <div
        className="MuiChip-root chip-email MuiChip-colorPrimary"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
      >
        <div
          className="MuiAvatar-root MuiAvatar-circular MuiChip-avatar MuiChip-avatarColorPrimary MuiAvatar-colorDefault"
          color="primary"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
            />
          </svg>
        </div>
        <span
          className="MuiChip-label"
        >
          separate@knob
        </span>
      </div>
    </div>
    <div
      className="makeStyles-profileInfoArea-4"
    >
      <hr
        className="MuiDivider-root"
      />
      <div
        className="label-info-container"
      >
        <label
          className="control-label label-block"
        >
          Identity Provider
        </label>
        <label
          className="label-normal"
        >
          monks
        </label>
      </div>
      <div
        className="label-info-container"
      >
        <label
          className="control-label label-block"
        >
          Last Login
        </label>
        <label
          className="label-normal"
        >
          May 5th, 2018 12:00:00 am
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`NeonAuth - UserCard Renders without a lastLogin prop 1`] = `
<div
  className="makeStyles-profileArea-3"
>
  <div
    className="makeStyles-card-2"
  >
    <div
      className="makeStyles-cardHeaderContainer-1"
    >
      <img
        alt="Profile"
        className="profile-image"
        src="carhole"
      />
    </div>
    <div>
      <h4
        className="MuiTypography-root MuiTypography-h4"
        style={
          Object {
            "margin": "10px",
          }
        }
      >
        Guy Incognito
      </h4>
    </div>
    <div
      className="makeStyles-emailChipContainer-5"
    >
      <div
        className="MuiChip-root chip-email MuiChip-colorPrimary"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
      >
        <div
          className="MuiAvatar-root MuiAvatar-circular MuiChip-avatar MuiChip-avatarColorPrimary MuiAvatar-colorDefault"
          color="primary"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
            />
          </svg>
        </div>
        <span
          className="MuiChip-label"
        >
          joeyjojojr@shabadu
        </span>
      </div>
    </div>
    <div
      className="makeStyles-profileInfoArea-4"
    >
      <hr
        className="MuiDivider-root"
      />
      <div
        className="label-info-container"
      >
        <label
          className="control-label label-block"
        >
          Identity Provider
        </label>
        <label
          className="label-normal"
        >
          momar
        </label>
      </div>
      <div
        className="label-info-container"
      >
        <label
          className="control-label label-block"
        >
          Last Login
        </label>
        <label
          className="label-normal"
        />
      </div>
    </div>
  </div>
</div>
`;
